dotNETStandard/HealthVault.Mobile/cs/Xamarin.Core/ViewModels/MedicationEditViewModel.cs

```diff
@@ -6,38 +6,46 @@
 using System.Windows.Input;
 using HealthVault.Sample.Xamarin.Core.Services;
 using Microsoft.HealthVault.Clients;
+using Microsoft.HealthVault.Connection;
 using Microsoft.HealthVault.ItemTypes;
+using Microsoft.HealthVault.Person;
 using Microsoft.HealthVault.Vocabulary;
 using Xamarin.Forms;
 
 namespace HealthVault.Sample.Xamarin.Core.ViewModels
 {
     public class MedicationEditViewModel : ViewModel
     {
+        private readonly Medication medication;
+        private readonly IHealthVaultConnection connection;
+
         public MedicationEditViewModel(
             Medication medication,
-            IList<VocabularyItem> ingredientChoices,
-            IThingClient thingClient,
-            Guid recordId, 
+            IHealthVaultConnection connection,
```

Comment:
It's often better to inject a few more objects for the benefit of simpler dependency graph, which simplifies testing etc.  
Here's a good video on the subject [[https://www.youtube.com/watch?v=acjvKJiOvXw](https://www.youtube.com/watch?v=acjvKJiOvXw)]([https://www.youtube.com/watch?v=acjvKJiOvXw](https://www.youtube.com/watch?v=acjvKJiOvXw))

However, in our case we have multiple nested ViewModels so we can end up injecting most of the clients anyways. 


---
In reply to: [117591813](https://github.com/Microsoft/healthvault-samples/pull/11#discussion_r117591813) [](ancestors = 117591813)